File: frontend/src/apps/todo/models/todo.js
import moment from "moment";
import { uuid, assign } from "@/utils/Objects";

export class TodoItemStatus {
  static InProgress = "in_progress";
  static Done = "done";
}

export class TodoItemPriority {
  static Urgent = 4;
  static High = 3;
  static Medium = 2;
  static Low = 1;
}

export class TodoItem {
  constructor(data) {
    this.__uuid__ = uuid();
    this.id = null;
    this.name = "";
    this.description = "";
    this.priority = TodoItemPriority.Low;
    this.status = TodoItemStatus.InProgress;
    this.created_at = moment();
    this.updated_at = moment();
    this.assign(data);
  }

  assign(data) {
    const __uuid__ = this.__uuid__;
    assign(this, data, { __uuid__ });
    typeof this.created_at === "string" && (this.created_at = moment(this.created_at));
    typeof this.updated_at === "string" && (this.updated_at = moment(this.updated_at));
  }
}

File: backend/todo/models/todo.py
from django.db import models
from django.urls import reverse
from django.utils.translation import gettext as _

# pylint: disable=line-too-long
class TodoItem(models.Model):

    # pylint: disable=invalid-name
    class Priority(models.IntegerChoices):
        Urgent = 4, _("Urgent")
        High = 3, _("High")
        Medium = 2, _("Medium")
        Low = 1, _("Low")

    # pylint: disable=invalid-name
    class Status(models.TextChoices):
        InProgress = 'in_progress', _("In Progress")
        Done = 'done', _("Done")

    name = models.CharField(null=False, blank=False, default="", max_length=255)
    description = models.CharField(null=False, blank=False, default="", max_length=255)
    priority = models.SmallIntegerField(null=False, blank=False, choices=Priority.choices, default=Priority.Low)
    status = models.CharField(null=False, blank=False, choices=Status.choices, default=Status.InProgress, max_length=16)

    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)
    deleted_at = models.DateTimeField(null=True, default=None)

    class Meta:
        verbose_name = _("Todo Item")
        verbose_name_plural = _("Todo Items")

    # pylint: disable=invalid-str-returned
    def __str__(self):
        return self.name

    def get_absolute_url(self):
        return reverse("TodoItem-details", kwargs={"pk": self.pk})
